src/lcm/input_processing/discrete_grid_conversion.py

```diff
@@ -56,51 +66,57 @@ def params_to_internal(self, params: ParamsDict) -> ParamsDict:
 
         """
         out = params.copy()
-        for var in self.converted_states:
-            out.pop(f"next_{var}")
-            out[f"next___{var}_index__"] = params[f"next_{var}"]
+        for var in self.index_to_code:
+            old_name = f"next_{var}"
+            if old_name in out:
+                out[f"next___{var}_index__"] = out.pop(old_name)
         return out
 
-    def internal_to_states(self, states: dict[str, Array]) -> dict[str, Array]:
+    def internal_to_discrete_vars(
```

Comment:
Would prefer `internal_vars_to_vars` and `vars_to_internal_vars`, because with `internal_vars_to_discrete_vars` I get the impression that all internal variables are converted to discrete variables.

Reply:
Great.